include/dlaf/eigensolver/backtransformation/mc.h

```diff
@@ -179,6 +191,166 @@ void BackTransformation<Backend::MC, Device::CPU, T>::call_FC(
   }
 }
 
+template <class T>
+void BackTransformation<Backend::MC, Device::CPU, T>::call_FC(
+    comm::CommunicatorGrid grid, Matrix<T, Device::CPU>& mat_c, Matrix<const T, Device::CPU>& mat_v,
+    common::internal::vector<hpx::shared_future<common::internal::vector<T>>> taus) {
+  using hpx::unwrapping;
+
+  using hpx::execution::parallel_executor;
+  using hpx::resource::get_thread_pool;
+  using hpx::threads::thread_priority;
+
+  auto executor_hp = dlaf::getHpExecutor<Backend::MC>();
+  auto executor_np = dlaf::getNpExecutor<Backend::MC>();
+
+  // Set up MPI
+  auto executor_mpi = dlaf::getMPIExecutor<Backend::MC>();
+  common::Pipeline<comm::Communicator> mpi_col_task_chain(grid.colCommunicator());
+  common::Pipeline<comm::Communicator> mpi_row_task_chain(grid.rowCommunicator());
+
+  const SizeType c_local_rows = mat_c.distribution().localNrTiles().rows();
+  const SizeType c_local_cols = mat_c.distribution().localNrTiles().cols();
+  const SizeType v_local_rows = mat_v.distribution().localNrTiles().rows();
+  const SizeType m = mat_c.nrTiles().rows();
+  const SizeType n = mat_c.nrTiles().cols();
+  const SizeType mb = mat_v.blockSize().rows();
+
+  auto dist_v = mat_v.distribution();
```

Comment:
minor suggestion: it is possible to use `dist_v` in the code to make it slightly more readable. you can also think about doing the same thing with `mat_c.distribution()`.